### PR TITLE
label: remove the needs-cherry-pick labels for archived doc versions

### PR DESCRIPTION
**Pull request by qiancai**

### CODE REVIEW

File: prow/config/labels.md [hunk 1]
```diff
@@ -31,10 +31,6 @@
 - [Labels that apply to pingcap/tiflow, only for PRs](#labels-that-apply-to-pingcaptiflow-only-for-prs)
 - [Labels that apply to ti-community-infra/configs, for both issues and PRs](#labels-that-apply-to-ti-community-infraconfigs-for-both-issues-and-prs)
 - [Labels that apply to ti-community-infra/configs, only for issues](#labels-that-apply-to-ti-community-infraconfigs-only-for-issues)
-- [Labels that apply to ti-community-infra/devstats, for both issues and PRs](#labels-that-apply-to-ti-community-infradevstats-for-both-issues-and-prs)
```
Comment (qiancai) [inline prow/config/labels.md:34]: This was deleted by scripts/update-labels.sh. Not sure if it is correct. @purelind PTAL Thanks.

**Review by ran-huang — APPROVED** [review]

File: prow/config/labels.md (hunk 1 above)
Comment (wuhuizuo) [reply prow/config/labels.md:34]: it's ok, the repo is archived.

**Pull request merged**